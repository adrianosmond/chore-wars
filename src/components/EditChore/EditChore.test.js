import React from 'react';
import { shallow } from 'enzyme';
import { EditChore } from './index';

const testTime = new Date('2018-02-18').getTime();
const chores = {
  'test-chore': {
    title: 'Test Chore',
    frequency: 1,
    pointsPerTime: 100,
    lastDone: 0,
  },
};
const match = { params: { slug: 'test-chore' } };

describe('Edit Chore', () => {
  it('Renders', () => {
    expect(shallow(<EditChore match={match} chores={chores} currentTime={testTime} />)).toMatchSnapshot();
  });

  it('Should try to load the chores if they havent already been loaded', () => {
    const mockLoadChores = jest.fn();
    expect(shallow(<EditChore match={match} loadChores={mockLoadChores} currentTime={testTime} />)).toMatchSnapshot();
    expect(mockLoadChores).toHaveBeenCalled();
  });
});
