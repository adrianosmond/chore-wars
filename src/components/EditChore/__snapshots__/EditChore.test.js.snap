// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Chore Renders 1`] = `
<ChoreForm
  chore={
    Object {
      "frequency": 1,
      "lastDone": 0,
      "pointsPerTime": 100,
      "title": "Test Chore",
    }
  }
  currentTime={1518912000000}
  onSubmit={[Function]}
/>
`;

exports[`Edit Chore Should try to load the chores if they havent already been loaded 1`] = `""`;
